### PR TITLE
Add: (quelpa-upgrade-all)

### PR DESCRIPTION
**Pull request by alphapapa**
Split from #152.

### CODE REVIEW

File: quelpa.el [hunk 1]
```diff
@@ -1784,6 +1784,23 @@ the `quelpa' command has been run in the current Emacs session."
...
+  "Upgrade a package found in `quelpa-cache' with recipe RCP."
+  (interactive
+   (when (quelpa-setup-p)
+     (let* ((quelpa-melpa-recipe-stores (list quelpa-cache))
+            (name (quelpa-interactive-candidate))
+            (prefix (prefix-numeric-value current-prefix-arg)))
+       (list (assoc name quelpa-cache)))))
+  (when rcp
+    (let* ((quelpa-upgrade-p t)
+           (current-prefix-arg nil))
```
Comment (alphapapa) [inline quelpa.el]: As we discussed in the other PR, all the interactive stuff should be handled in the `interactive` form, so `current-prefix-arg` shouldn't be touched here.

File: quelpa.el [hunk 2]
```diff
@@ -1784,6 +1784,23 @@ the `quelpa' command has been run in the current Emacs session."
...
+      (setq quelpa-cache
+            (cl-remove-if-not (lambda (item) (package-installed-p (car item))) quelpa-cache))
```
Comment (alphapapa) [inline quelpa.el]: One-line lambdas can be difficult to read.  :)

**Comment by kiennq** [comment]
Sure, please go ahead and make changes accordingly.

File: quelpa.el (hunk 1 above)
Comment (kiennq) [reply quelpa.el]: `(quelpa rcp)` is actively checking `current-prefix-arg`, do you want to change behavior of that function too?

Comment (alphapapa) [reply quelpa.el]: I think you mean this binding in `quelpa`:

```el
(quelpa-upgrade-p (if current-prefix-arg t quelpa-upgrade-p))
```
If `current-prefix-arg` is nil, it falls back to `quelpa-upgrade-p`, which is still being bound.

**Pull request merged**